docs/workflow_tasks/discover_granules.md

```diff
@@ -78,6 +78,16 @@ filename occurs as described above.
 When set on the task configuration, the value applies to all collections during
 discovery.  Otherwise, this property may be set on individual collections.
 
+#### Concurrency
+
+A number property that determines the level of concurrency with which granule duplicate checks are performed when `duplicateGranuleHandling` is `skip` or `error`.
+
+Limiting concurrency helps to avoid throttling by the AWS Lambda API and helps to avoid encountering account Lambda concurrency limitations.
+
+We do not recommend increasing this value to a level at which you may encounter throttling errors, but if you are seeing Lambda.Timeout errors when discover-granules discovers a large number of granules with `skip` or `error` duplicate handling, you may either wish to increase this value or convert the discover-granules task to an ECS activity, which does not face similar runtime constraints.
```

Comment:
Nit: 

This might be better said if we went with something like *emphasis on like*:  "We do not recommend increasing this value unless you are seeing Lambda.Timeout errors when running this task against a large number of granules with `skip` or `error` duplicate handling.   If you do, this value allows for tuning, but be aware that larger values increase the potential for overwhelming the AWS Lambda API for low values of `Concurrent Executions`, as each granule requires an API lambda invocation to check for granule status.

In short, I think we should put the reason for tweaking this first, with the scaling concern caveat last, and reference briefly why it's a concern.